aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1000,8 +1110,19 @@ static void apply_svd(Tensor& self, Tensor& U, Tensor& S, Tensor& VT,
   int info;
   auto m = self.size(-2);
   auto n = self.size(-1);
-  auto k = std::min(m, n);
-  Tensor iwork = at::empty({8 * k}, at::kInt);
+  auto mn = std::min(m, n);
+  auto mx = std::max(m, n);
+  int64_t lrwork; // These settings are based on LAPACK 3.8.
+  if (jobz == 'N'){
+    lrwork = 5*mn;
+  }else if (mx > 10*mn){
+    lrwork = 5*mn*mn + 5*mn;
+  } else {
+    lrwork = std::max(5*mn*mn + 5*mn, 2*mx*mn + 2*mn*mn + mn);
+  }
+  Tensor rwork = at::empty({std::max(int64_t(1), lrwork)}, at::kInt);
+  auto rwork_data = rwork.data_ptr<int>();
+  Tensor iwork = at::empty({8*mn}, at::kInt);
```

Comment:
Could you please include spaces before and after `*`?
Couple more comments:
1. For LAPACK 3.6, `lrwork` needs to be at least `7 * mn`. Could we do that so that users with older LAPACK versions are not at a disadvantage?
1. Also, in the case `mx > 10 * mn`: how did you obtain `10`?

Reply:
1. I have made the changes and update the comment to say it supports LAPACK 3.6.
2. I had to make the assumption that `mx >> mn` implies `mx > 10 * mn`.  I don't support `torch.rand()` with complex numbers so I don't really have a good way to test this right now. 

Once `legacy::cpu::_th_random_` is ported from TH I should be able to better test this. Also, rwork only impacts complex number svd calls.

Reply:
Just one more concern: if rwork is required for complex valued matrices / tensors, why allocate it for the general case?

Can't we special case as follows?
```c++
Tensor rwork;
int* rwork_data = std::nullptr;
if (at::isComplex(self)) {
  rwork = at::empty(.., at::kInt);
  rwork_data = rwork.data_ptr<int>();
}
```

Reply:
Good point! I will update the code.

Reply:
Ok, I've made the changes. Have a good weekend.

Reply:
Thank you @dylanbespalko, you have a good one too.